src/pages/ultimate.tsx

```diff
@@ -0,0 +1,256 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { css } from 'emotion';
+
+export enum Winner {
+  Nought,
+  Cross,
+  Draw,
+}
+
+export interface Cell {
+  player: Nullable<Winner>;
+}
+
+export interface Board {
+  cells: Cell[];
+}
+
+function calculateWinner(squares: Cell[]): Nullable<Winner> {
+  const lines = [
+    [0, 1, 2],
+    [3, 4, 5],
+    [6, 7, 8],
+    [0, 3, 6],
+    [1, 4, 7],
+    [2, 5, 8],
+    [0, 4, 8],
+    [2, 4, 6],
+  ];
+  for (const [a, b, c] of lines) {

```

Comment:
```js
const winningLine = lines.find(([a, b, c]) => squares[a].player !== null && ...);
return winningLine ? squares[winningLine[0]].player
  : squares.every(_.identity) ? Winner.Draw
  : null
```

(Double ternary optional; `if (winningLine) { ... } else { return squares.every(...) ? ... }` is also fine.)